### ISSUE TITLE
npm run new fails when VS Code CLI is not installed

### ISSUE THREAD

**Issue by 2nofa11** [vuejs/eslint-plugin-vue#2781]
---
name: Bug report
about: Create a report to help us improve
---

**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.17.0 (development environment issue, not rule-specific)
- **eslint-plugin-vue version:** 10.2.0
- **Vue version:** 3.x (development environment issue, not app-specific)
- **Node version:** v23.11.0
- **Operating System:** macOS (Darwin 24.5.0)

**Please show your full configuration:**
This is a development tooling issue with npm run new command.

**What did you do?**
```bash
npm run new <rule-name> <author>
```

**What did you expect to happen?**

The command should complete successfully without errors, providing a smooth experience for new contributors following the developer guide.

**What actually happened?**

The command generates files successfully but fails at the end when trying to automatically open files in VS Code:

```
/bin/sh: code: command not found
Error: Command failed: code "path/to/generated/file"
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at checkExecSyncError (node:child_process:882:11)
    at Object.execSync (node:child_process:954:15)
    at /path/to/tools/new-rule.js:143:6
```

The issue occurs because the script assumes VS Code CLI (`code` command) is installed, but many developers use other editors (like Cursor, Vim, etc.) or don't have VS Code command line tools available. This creates a poor first impression for new contributors.

**Repository to reproduce this issue**

This issue can be reproduced on any system where VS Code command line tools are not installed. The problem is in `tools/new-rule.js:143-145` where it executes `code` command without error handling to open generated files automatically.

**Suggested Solution**

Add graceful fallback when `code` command is not available:
```javascript
try {
  cp.execSync(`code "${ruleFile}"`, { stdio: 'ignore' })
} catch (err) {
  // Continue silently if VS Code is not available
  console.log('Files generated successfully')
}
```

**Comment by ValentinYoushkevich** [comment]
Opened #3116 for this.

Root cause is exactly where you pointed — [`tools/new-rule.js:142-144`](https://github.com/vuejs/eslint-plugin-vue/blob/master/tools/new-rule.js#L142-L144):

```js
cp.execSync(`code "${ruleFile}"`)
cp.execSync(`code "${testFile}"`)
cp.execSync(`code "${docFile}"`)
```

All three files are written before this point, so the generation itself always succeeds. The editor launch is what fails, and because the first `execSync` throws, the other two files are never opened either, and the paths of the generated files are never printed — so the failure looks worse than it is.

Reproduced on Windows too, with `code` removed from `PATH`:

```
'code' is not recognized as an internal or external command
Error: Command failed: code "...\lib\rules\__probe.ts"
    at Object.execSync (node:child_process:954:15)
    at tools\new-rule.js:142:6
EXITCODE=1
```

The PR prints the generated paths and ignores a failure to launch the editor:

```
$ npm run new some-rule Someone
Created:
  lib/rules/some-rule.ts
  tests/lib/rules/some-rule.test.ts
  docs/rules/some-rule.md
$ echo $?
0
```

With `code` available the three files are still opened as before.

Not covered on purpose: the PR does not add support for `$EDITOR`/`$VISUAL` or for other editor CLIs (Cursor, Vim, …). It only makes the command exit cleanly when `code` is missing, which keeps the diff to the reported bug. Happy to add editor detection if maintainers want it.
